### PR TITLE
[DEV-161] Init 2FA Guru doc

### PR DESCRIPTION
**Pull request by smilenow**

### CODE REVIEW

File: docs/2fa/introduction.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# Ready to start?
+
+As a user, if you're interested in our 2FA protected and want to have a more secured assets, what you need to prepare for 2FA are:
+
+- Primary wallet, the place where you store and control your assets.
+
+- Recovery wallet, it's used for reset the 2FA device if you lost it.
+
+- 2FA device, retrieve your 2FA TOTP code before you make any authorised transfer, please don't use the same device as the primary wallet's.
```
Comment (Liao1) [inline docs/2fa/introduction.md:31]: retrieve your 2FA TOTP(Time-based One-time Password)

File: docs/2fa/users/guide.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+---
+
+## Setup 2FA
+<!-- - Visit [Automata 2FA Guru](https://app.2fa.guru). -->
+- Visit Automata 2FA Guru.
+
+- Use metamask and connect your wallet.
+
+- If your account hasn't signed up, you can click the **Wallet Actions** button on the top, and go through the signup process. You need to sign a message to prove your identity before generating a 2FA secret. Attention, you must use a different device to store the 2FA secret, don't keep the 2FA device and your wallet in the same one.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:14]: If you haven't signed up for an account, you can click the **Wallet Actions** button at the top to initiate the process of signing up. You will be required to sign a message to prove your identity before generating a 2FA secret. Take note, the primary wallet and 2FA device should be different devices."

File: docs/2fa/users/guide.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+![](../../assets/2fa/setup-1.png)
+
+![](../../assets/2fa/setup-2.png)
+
+![](../../assets/2fa/setup-3.png)
+
+And we also support 1password for the 2FA secret representation. You need to prepare a **Login** item in 1password, configure the basic information, then you can use **Scan QR Code** function in 1password. It will make it more flexiable and efficient.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:22]: We support 1Password for storing your 2FA secret. To enable the use of **Scan QR Code** function in 1Password, you would first have to prepare a **Login** item in 1Password and configure the basic information. This makes password retrieval more convenient.

File: docs/2fa/users/guide.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+![](../../assets/2fa/1password.png)
+
+Recommend configuration in 1password
```
Comment (linyao7) [inline docs/2fa/users/guide.md:26]: Recommend configuration in 1Password

File: docs/2fa/users/guide.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+| | |
+| --- | --- |
+| username | wallet address |
+| website | https://app.2fa.guru |
+| tags | the target network |
+
+- Once you successfully sign up, we recommend you add a recovery wallet for your primary account, which can be used to reset your 2FA device. Otherwise, if you lost your 2FA device, then the only thing you can do is to disable the 2FA protection before you use your 2FA protected assets.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:34]: Once you've successfully signed up, we recommend that you add a recovery wallet for your primary wallet, which can be used to reset your 2FA device. Without a recovery wallet, you can only disable the 2FA protection before you use your 2FA protected assets.

File: docs/2fa/users/guide.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  You need to send a bind transaction to our 2FA management contract after our authentication, which means the gas fee is required.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:36]: For binding a recovery wallet, you would need to send a binding transaction to our 2FA management contract after our authentication, and gas fees would be required.

File: docs/2fa/users/guide.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+  There is a binding window for each account after signup, you must bind the recovery wallet in **7 days**, otherwise you cannot bind anymore.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:38]: There is a binding window for each account after signup. If you do not bind a recovery wallet in **7 days**, you would not be allowed to bind one subsequently.

File: docs/2fa/users/guide.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+![](../../assets/2fa/bind-1.png)
+
+![](../../assets/2fa/bind-2.png)
+
+---
+
+## Reset 2FA devices
+
+If you lost your 2FA device unfortunately, you can reset it if you have been bind a recovery wallet, the reset processes are quite similar as setup's:
```
Comment (linyao7) [inline docs/2fa/users/guide.md:48]: If you lost your 2FA device in an unfortunate incident, you can reset it only if you have a recovery wallet. The reset processes are similar to setting up 2FA for the first time:

File: docs/2fa/users/guide.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+<!-- - Visit [Automata 2FA Guru](https://app.2fa.guru). -->
+- Visit Automata 2FA Guru.
+
+- Use metamask and connect your recovery wallet.
+
+- Click **Reset 2FA** button in **Wallet Actions**, input your primary wallet address and sign the message. If it's matched, then you can scan the new secret to reset the 2FA device. There is no transaction on chain, so it's free to reset a 2FA device. And the recovery wallet is not required to sign up for 2FA.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:55]: Click **Reset 2FA** button in **Wallet Actions**, input your primary wallet address and sign the message. If the address pair matches, you can scan the new secret to reset the 2FA device. There is no transaction on chain, so no gas fees are required to reset a 2FA device. Take note, the recovery wallet does not need to sign up for 2FA for this to work.

File: docs/2fa/users/guide.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+![](../../assets/2fa/reset-1.png)
+
+![](../../assets/2fa/reset-2.png)
+
+---
+
+## Make authorised transfer
+
+Once you sign up 2FA for your wallet, you will benefit from all 2FA protected assets and dApps. If you get any 2FA protected assets, you will be able to transfer them in **Manage my Wallet Assets** panel.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:65]: Once you sign up 2FA for your wallet, your 2FA assets in supported dApps will be protected. If you hold any 2FA protected assets, you will be able to transfer them in **Manage my Wallet Assets** panel.

File: docs/2fa/users/guide.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+![](../../assets/2fa/transfer-2.png)
+
+![](../../assets/2fa/transfer-3.png)
+
+---
+
+## Disable 2FA protection
+
+If you lost your 2FA device and don't bind any recovery wallet, or you don't want to use 2FA anymore, you can disable 2FA protection in 2FA dApp. There is a cooldown period when disabling 2FA protection, you can cancel the disable 2FA protection request before this window. Attention, the cooldown period is **7 Days**, and you cannot use **2FA protection** anymore.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:79]: If you've lost your 2FA device and no respective recovery wallet has been bound, you always have the option to disable 2FA protection in the 2FA dApp. There is a cooldown period for disabling 2FA protection, and you can cancel the disable 2FA protection request before this period ends. The cooldown period is **7 Days**, and after which you would not be able to use **2FA protection** anymore.

File: docs/2fa/users/guide.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+
+So it's recommended to transfer all your 2FA protected assets to another account after you disable the 2FA, and sign up the 2FA protection for this new account as soon as possible, to make sure all assets are safe.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:81]: Hence, it's recommended to transfer all your 2FA protected assets to another 2FA protected account after you've disabled the 2FA for a primary wallet. This ensures that all assets will stay protected.

File: docs/2fa/users/guide.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+### Steps to disable 2FA
+
+<!-- - Visit [Automata 2FA Guru](https://app.2fa.guru). -->
+- Visit Automata 2FA Guru.
+
+- Use metamask and connect your wallet.
+
+- Click **Disable 2FA** button in **Wallet actions**, sign the message and wait the cooldown window ends. Attention, the 2FA is still active during this period, and all authorised operations are required.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:90]: Click **Disable 2FA** button in **Wallet actions**, sign the message and wait until the cooldown window ends. Do note, 2FA is still active during this period, and all operations will require authorisation.

File: docs/2fa/users/guide.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+
+![](../../assets/2fa/disable.png)
+
+![](../../assets/2fa/disabling.png)
+
+If you trigger the disable procedure by mistake, you can cancel it during the cooldown window to keep the 2FA protection.
+
+- Click **Cancel Disable 2FA** button in **Wallet actions**, sign the message and you can use the previous 2FA device to get the 2FA protection.
```
Comment (linyao7) [inline docs/2fa/users/guide.md:98]: Click **Cancel Disable 2FA** button in **Wallet actions**, sign the message and this would cancel the request to disable 2FA on the current wallet. You can then continue to use the previous 2FA device with 2FA protection.

File: docs/2fa/developers/faq.md [hunk 15]
```diff
@@ -0,0 +1,5 @@
+# Frequently Asked Questions
+
+### **1. How can I integrate with the Automata 2FA protection?**
```
Comment (linyao7) [inline docs/2fa/developers/faq.md:3]: ### **1. How can I integrate my Dapp with Automata's 2FA protection?**

File: docs/2fa/developers/faq.md [hunk 16]
```diff
@@ -0,0 +1,5 @@
...
+
+You can refer to smart contract part to see how to integrate with your contract, please make sure all assets ownership transfer operations should add **only2FA** modifier.
```
Comment (linyao7) [inline docs/2fa/developers/faq.md:5]: You can refer to the smart contract section for more technical details. Do ensure that all asset transfer operation include a **only2FA** modifier.

File: docs/2fa/developers/workflow.md [hunk 17]
```diff
@@ -0,0 +1,27 @@
+# Workflow
+
+![](../../assets/2fa/workflow.png)
+
+### Key Components
+* Primary wallet:
+
+  The place where users store and control their assets.
```
Comment (linyao7) [inline docs/2fa/developers/workflow.md:8]: The wallet where users store and control their assets.

File: docs/2fa/developers/workflow.md [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+
+* Recovery wallet:
+
+  Used for identifying the users' roles and resetting the 2FA device, the relationship between the primary wallet and the recovery wallet is saved on-chain after binding.
```
Comment (linyao7) [inline docs/2fa/developers/workflow.md:12]: Used for identifying users' roles and resetting the 2FA device. The relationship between the primary wallet and the recovery wallet is saved on-chain after binding.

File: docs/2fa/developers/workflow.md [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+
+* 2FA client:
+
+  The device which provides the TOTP(Time-based one-time password).
+
+<!-- * [Automata 2FA Guru](https://app.2fa.guru) -->
+* Automata 2FA Guru
+
+  The dApp hosted by the Automata team. It contains the 2FA management panel and wallet assets control panel. Users can sign up 2FA and find the 2FA protected dApps in the management panel, and they can make the authorised transfer under the wallet assets control panel.
```
Comment (linyao7) [inline docs/2fa/developers/workflow.md:21]: The dApp hosted by the Automata team. It contains the 2FA management and wallet assets control panels. Users can sign up for 2FA and find the 2FA protected dApps in the management panel. They can also make authorised transfers under the wallet assets control panel.

File: docs/2fa/developers/workflow.md [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+
+* 2FA Guru Geode: 
+
+  The 2FA middleware which is built upon the [Intel SGX](https://medium.com/atanetwork/introduction-to-sgx-the-eli5-3abf25e50057). With the provision of a secure, fully-shielded enclave, the platform connects seamlessly with dApps to provide support for 2FA verification on public blockchains. It's responsible for the 2FA secret generation, storage and authentication. The attackers cannot predict or calculate the valid signature outside the enclave, and its storage is encrypted by the enclave and the attackers cannot view anything even if they steal the data.
```
Comment (linyao7) [inline docs/2fa/developers/workflow.md:25]: The 2FA middleware which is built upon the [Intel SGX](https://medium.com/atanetwork/introduction-to-sgx-the-eli5-3abf25e50057). With the provision of a secure, fully-shielded enclave, the platform connects seamlessly with dApps to provide support for 2FA verification on public blockchains. It's responsible for the 2FA secret generation, storage and authentication. Attackers cannot predict or calculate the valid signature outside the enclave, and its storage is encrypted internally and attackers are unable to view anything even if they managed to steal data.

File: docs/2fa/developers/workflow.md [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+
+  Even though the assets are safe under the 2FA protection, it's still possible for attackers to steal the TOTP by using the phishing attack, and transfer the victims' 2FA protected assets. We introduce anti-phishing functionality to prevent it, our 2FA middleware can analyze the original domain and detect whether it's coming from the phishing domains or links by using the daily updated phishing blacklists. It will reject to provide a valid signature and keep users' assets safe. To meet the highest security, we also support a request domain whitelist for a specified dApp.
```
Comment (linyao7) [inline docs/2fa/developers/workflow.md:27]: Even though the assets are safe under the 2FA protection, it's still possible for attackers to steal the TOTP by using a phishing attack. To prevent this, we use an anti-phishing feature where our 2FA middleware can analyse the original domain and detect whether it's coming from the phishing domains or links by using daily updated phishing blacklists. If a malicious domain is detected, the geode will not provide a signature to keep users' assets safe. To uphold high security standards, we also support a custom domain whitelist for a specified dApp.

File: docs/2fa/users/guide.md (hunk 5 above)
Comment (smilenow) [reply docs/2fa/users/guide.md:34]: I will refine it to `Without a recovery wallet, if you lost your 2FA device, then you can only disable the 2FA protection before you use your 2FA protected assets.` for the last sentence.

**Review by Gnnng — APPROVED** [review]
